src/commonMain/kotlin/com/github/quillraven/fleks/entity.kt

```diff
@@ -97,6 +97,27 @@ open class EntityCreateContext(
         val holder: ComponentsHolder<T> = componentService.holder(compType)
         holder[this] = component
     }
+
+    /**
+     * Adds the [components] to the [entity][Entity]. This function should only be used
+     * in exceptional cases.
+     * It is preferred to use the [plusAssign] function whenever possible to have type-safety.
+     *
+     * If a component [addHook][ComponentsHolder.addHook] is defined then it
+     * gets called after the component is assigned to the [entity][Entity].
+     *
+     * If a component [removeHook][ComponentsHolder.removeHook] is defined and the [entity][Entity]
+     * already had such a component then it gets called with the previous assigned component before
+     * the [addHook][ComponentsHolder.addHook] is called.
+     */
+    inline operator fun Entity.plusAssign(components: List<Component<*>>) {
```

Comment:
I think this will be useful. Since I am currently playing with serialization of entity and components. It will be very useful if one want to update "unspecific" amount of components of an entity. This will just be easier than if in that case you need to check for each component type by yourself.

/offtopic
BTW I tried again Kotlinx.serialization with a world snapshot and I got a bit further. I need to "re-map" the map from the snapshot, but I got around the "Any" and the Star projection problem :)

```json
{
    "entity": "42",
    "components": [
        {
            "type": "TestComponent",
            "payload": {
                "x": 42.1,
                "y": 17.3
            }
        }
    ]
}
```

Reply:
Anything we can do to improve that to maybe avoid the re-mapping?

Reply:
It is OK for now and I won't change the map of snapshot().

I don't know if I will change the structure again when I find an easier setup which can be serialized. The problem is not the entity as key, but the list of `Component<*>`. First I need to cast it to `Component<Any>`. For `Component<Any>` I have written a KSerializer strategy. "payload" is of type `Componenty<Any>` but it lost its underlying type. So I need that from "type" as string. In the `KSerializer<Component<Any>>` the string is than mapped to the actual (default) serializer of the final component (here: `TestComponent(...) : Component<Any> {...}`. This is totally weird but it works.
 
So -after thinking about it- we could change `snapshot(): Map<Entity, List<Component<*>>>` to `snapshot(): Map<Entity, List<Component<Any>>>` -> `*` to `Any` :)
I will continue to look into my code and maybe I find a way to directly serialize the returned map.

Reply:
I just read your edit now AFTER I adapted the functions haha.

I had to make it Component<out Any> instead of Component<Any> because otherwise some type interference did not work properly.

But the change is no longer needed if I understand you correctly? Then I'd prefer to revert it.

Reply:
Yes, you can revert it. I found a serializer solution which is in principle totally independent of Component<*> type. I will update my korge-fleks repo the next days for anyone who is interested:)